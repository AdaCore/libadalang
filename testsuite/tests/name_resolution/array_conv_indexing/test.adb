procedure Test is
   type Array_Type is array (Integer range <>) of Integer;

   Arr : Array_Type (1 .. 10);
   I : Integer;
begin
   I := Array_Type (Arr) (I);
   pragma Test_Statement;
end Test;

