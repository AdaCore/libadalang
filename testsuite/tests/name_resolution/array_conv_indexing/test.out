Analyzing test.adb
##################

Resolving xrefs for node <AssignStmt test.adb:7:4-7:30>
*******************************************************

Expr: <Id "I" test.adb:7:4-7:5>
  references:    <DefiningName "I" test.adb:5:4-5:5>
  type:          <ConcreteTypeDecl ["Integer"] __standard:4:3-4:54>
  expected type: None
Expr: <CallExpr test.adb:7:9-7:29>
  references:    <DefiningName "Array_Type" test.adb:2:9-2:19>
  type:          <ConcreteTypeDecl ["Integer"] __standard:4:3-4:54>
  expected type: <ConcreteTypeDecl ["Integer"] __standard:4:3-4:54>
Expr: <CallExpr test.adb:7:9-7:25>
  references:    <DefiningName "Array_Type" test.adb:2:9-2:19>
  type:          <ConcreteTypeDecl ["Array_Type"] test.adb:2:4-2:59>
  expected type: None
Expr: <Id "Array_Type" test.adb:7:9-7:19>
  references:    <DefiningName "Array_Type" test.adb:2:9-2:19>
  type:          None
  expected type: None
Expr: <Id "Arr" test.adb:7:21-7:24>
  references:    <DefiningName "Arr" test.adb:4:4-4:7>
  type:          <ConcreteTypeDecl ["Array_Type"] test.adb:2:4-2:59>
  expected type: None
Expr: <Id "I" test.adb:7:27-7:28>
  references:    <DefiningName "I" test.adb:5:4-5:5>
  type:          <ConcreteTypeDecl ["Integer"] __standard:4:3-4:54>
  expected type: <ConcreteTypeDecl ["Integer"] __standard:4:3-4:54>


Done.
